private-templates-service/client/src/components/Designer/Designer.tsx

```diff
@@ -12,45 +14,85 @@ import * as ACDesigner from 'adaptivecards-designer';
 const mapStateToProps = (state: RootState) => {
   return {
     isAuthenticated: state.auth.isAuthenticated,
-    user: state.auth.user
+    user: state.auth.user,
+    templateID: state.currentTemplate.templateID,
+    templateJSON: state.currentTemplate.templateID,
+    templateName: state.currentTemplate.templateName,
+    sampleDataJSON: state.currentTemplate.sampleDataJSON
   };
 };
 
+const mapDispatchToProps = (dispatch: any) => {
+  return {
+    updateTemplate: (templateID: string, templateJSON: string, templateName: string, sampleDataJSON: string) => {
+      dispatch(updateTemplate(templateID, templateJSON, templateName, sampleDataJSON));
+    }
+  }
+}
+
 interface DesignerProps {
   isAuthenticated: boolean;
   user?: UserType;
+  templateID: string;
+  templateJSON: string;
+  templateName: string;
+  sampleDataJSON: string;
+  updateTemplate: (templateID: string, templateJSON: string, templateName: string, sampleDataJSON: string) => any;
 }
 
-const Designer = (props: DesignerProps) => {
+let designer: ACDesigner.CardDesigner;
+
+class Designer extends React.Component<DesignerProps> {
+
+  componentWillMount() {
+    ACDesigner.GlobalSettings.enableDataBindingSupport = true;
+    ACDesigner.GlobalSettings.showSampleDataEditorToolbox = true;
+
+    ACDesigner.CardDesigner.onProcessMarkdown = (text: string, result: { didProcess: boolean, outputHtml?: string }) => {
+      result.outputHtml = new markdownit().render(text);
+      result.didProcess = true;
+    }
+
+    designer = initDesigner();
+
+    let publishButton = new ACDesigner.ToolbarButton("publishButton", "Publish", "", (sender) => (alert("Published!")));
```

Comment:
I'd be inclined to separate these out into separate functions (publish and save), which can be called here.